cmd/out/main.go

```diff
@@ -40,66 +44,84 @@ func main() {
 		request.Source.RetryDelay = 10 * time.Second
 	}
 
+	var (
+		executionTime  prometheus.Histogram
+		executionTimer *prometheus.Timer
+	)
+
+	if request.Source.PrometheusPushGateway != "" {
+		fmt.Fprintf(os.Stderr, "Setting up prometheus client, pushgateway=%s\n",
+			request.Source.PrometheusPushGateway)
+		executionTime = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Name: "resource_pool_execution_time",
+			Help: "Execution time of the current operation",
+		})
+		executionTimer = prometheus.NewTimer(executionTime)
+	} else {
+		fmt.Fprintln(os.Stderr, "No prometheus pushgateway set. Will not emit any metrics.")
+	}
+
 	lockPool := out.NewLockPool(request.Source, os.Stderr)
 
 	var (
-		lock    string
-		version out.Version
+		lock          string
+		version       out.Version
+		operationName string
 	)
 
+	// TODO bad design: should never support multiple operations in a single run
```

Comment:
All these changes (replacing if with else if) are changing the resource behavior. Before doing these, we need to validate with upstream that they are correct. To avoid being blocked while waiting for feedback, could you please remove these changes from this PR?

Reply:
Agreed, thanks.